Framework/Algorithms/src/CombineDiffCal.cpp

```diff
@@ -0,0 +1,170 @@
+  }
+
+  if (!isTableWorkspaceSortedById(pixelCalibrationWS)) {
+    results["PixelCalibration"] = "Please run SortTableWorkspace on the table submited as 'PixelCalibration'. ";
+  }
+
+  if (getTableWorspaceSortDirection(groupedCalibrationWS) != getTableWorspaceSortDirection(pixelCalibrationWS)) {
+    results["GroupedCalibration"] = "'GroupedCalibration's sort direction does not match 'PixelCalibration's sort "
+                                    "direction, flip one of their orders.";
+    results["PixelCalibration"] = "'PixelCalibration's sort direction does not match 'GroupedCalibration's sort "
+                                  "direction, flip one of their orders.";
+  }
+
+  return results;
+}
+
+std::shared_ptr<Mantid::API::TableRow> binarySearchForRow(DataObjects::TableWorkspace_sptr ws, int detid) {
+  size_t start = 0;
+  size_t end = ws->rowCount();
+  size_t currentPosition = end / 2;
+  int sortDirection = getTableWorspaceSortDirection(ws);
+  while (end - start > 0) {
+    Mantid::API::TableRow currentRow = ws->getRow(currentPosition);
+    if (currentRow.Int(0) * sortDirection > detid * sortDirection) {
+      end = currentPosition;
+    } else if (currentRow.Int(0) * sortDirection < detid * sortDirection) {
+      start = currentPosition;
+    } else {
+      return std::make_shared<Mantid::API::TableRow>(currentRow);
+    }
+    currentPosition = start + ((end - start) / 2);
+  }
+  return nullptr;
+}
+
+// Per Pixel:
+//
+// DIFC{eff} = (DIFC{pd}/DIFC{arb}) * DIFC{prev}
+//
+// DIFC{eff} = Output of this Alg, the combined DIFC
+// DIFC{pd} = The DIFC produced by PDCalibration, found in the "GroupedCalibration"
+// DIFC{arb} = found in the "CalibrationWorkspace" param
+// DIFC{prev} = The previous DIFCs, found in "PixelCalibration", as per description this was the set generated by CC
+
+//----------------------------------------------------------------------------------------------
+/** Execute the algorithm.
+ */
+void CombineDiffCal::exec() {
+  const DataObjects::Workspace2D_sptr calibrationWS = getProperty("CalibrationWorkspace");
+  const DataObjects::TableWorkspace_sptr groupedCalibrationWS = getProperty("GroupedCalibration");
+  const DataObjects::TableWorkspace_sptr pixelCalibrationWS = getProperty("PixelCalibration");
+
+  DataObjects::TableWorkspace_sptr outputWorkspace = std::make_shared<DataObjects::TableWorkspace>();
+  outputWorkspace->addColumn("int", "detid");
+  outputWorkspace->addColumn("double", "difc");
+  outputWorkspace->addColumn("double", "difa");
+  outputWorkspace->addColumn("double", "tzero");
+
+  Mantid::API::TableRow groupedCalibrationRow = groupedCalibrationWS->getFirstRow();
+  do {
+    std::shared_ptr<Mantid::API::TableRow> pixelCalibrationRow =
+        binarySearchForRow(pixelCalibrationWS, groupedCalibrationRow.Int(0));
+    if (pixelCalibrationRow) {
+      double value = (groupedCalibrationRow.Double(1) /
```

Comment:
Update this equation in a couple of ways.
1. Copy TZERO from previous calibration
2. DIFC is follows the equation already implemented
3. DIFA needs more time to sort out. Add an extra parameter `DIFAtolerance`, if the DIFA found in previous is smaller or equal, warn with the detector id and value, then set the value to zero. If the DIFA found in previous is larger, throw an exception.